src/utils/flags.ts

```diff
@@ -33,11 +40,115 @@ const parsePathInfo = async (input: string, exists: boolean): Promise<PathInfo>
   return { type: 'file', path: input };
 };
 
-export const fileOrDirFlag = (
-  opts: { exists?: boolean } & Partial<Interfaces.OptionFlag<PathInfo | undefined>> = {}
-): Interfaces.OptionFlag<PathInfo | undefined> => {
+type EnsureFsFlagOptions = {
+  flagName: string;
+  path: string;
+  type: 'dir' | 'file' | 'any';
+  throwOnENOENT?: boolean;
+};
+
+interface FsError extends Error {
+  code: string;
+}
+
+/**
+ * Ensures command flags that are file system paths are set properly before
+ * continuing command execution.  Can also create directories that don't yet
+ * exist in the path.
+ *
+ * @param options defines the path to resolve and the expectations
+ * @returns the resolved flag path
+ */
+function ensureFlagPath(options: EnsureFsFlagOptions): string {
+  const { flagName, path, type, throwOnENOENT } = options;
+
+  const trimmedPath = path?.trim();
+  let resolvedPath: string;
+  if (trimmedPath?.length) {
+    resolvedPath = resolve(trimmedPath);
+  }
+
+  try {
+    const stats = fs.statSync(resolvedPath);
+    if (type !== 'any') {
```

Comment:
why might it be `'any'` ?  If it is `'any'`, then nothing should happen?

Reply:
it's happy as long as something exists there?

Reply:
`any` denotes that it could be a file or a directory

For context, I took this directly from plugin-source: https://github.com/salesforcecli/plugin-source/blob/d8192203b018df2ed6af68ed695ca96e759a89c3/src/sourceCommand.ts#L81 without any modification

Reply:
if we don't have a use case for that in mind, I'd be happy omitting it